src/main/ruby/core/string.rb

```diff
@@ -287,12 +287,12 @@ def to_sub_replacement(result, match)
       while current < bytesize && data[current] != 92  # ?\\
         current += 1
       end
-      result.append(byteslice(index, current - index))
+      Truffle.invoke_primitive(:string_append, result, byteslice(index, current - index))
```

Comment:
Same here and below, since `String#<<` (or `concat` its alias) have the same semantics as `append` when passed a String, I'd prefer to use those standard primitives when we know the RHS is a String.

Reply:
You're right, we should have done a separate PR for this since the start, sorry.